### ISSUE TITLE
How to set rules for virtual category via API in Magento 2?

### ISSUE THREAD

**Issue by overjolted** [Smile-SA/elasticsuite#1298]
Hello, thank you for a great suite, that you guys have developed.
I do have a question though:

How do I create a virtual category and set rules for it via API?

PS To see what it might look like, I manually created a virtual category and set rules for it, but when I retrieve info about this category via API, 'virtual_rule' value is just an empty list. Why is that? Where do I find en example for setting rule for a virtual category via API?

Thanks.

### Preconditions

Magento Version :
Magento CE 2.3

ElasticSuite Version :
2.7.4

Environment :
Developer mode

Third party modules :
None

**Comment by romainruaud** [comment]
Hello @overjolted 

thank you for supporting us :) 

Just to be sure, do you mean "with the REST admin API" when speaking about "via API" ? 

Regards

**Comment by overjolted** [comment]
yes, I mean REST API

**Comment by overjolted** [comment]
> Hello @overjolted
> 
> thank you for supporting us :)
> 
> Just to be sure, do you mean "with the REST admin API" when speaking about "via API" ?
> 
> Regards

Yes, I mean REST API

**Comment by romainruaud** [comment]
For now I have implemented the **read** part. Attribute is now showing up like this in REST response : 

```
{
            "attribute_code": "virtual_rule",
            "value": {
                "condition": {
                    "condition_type": "Smile\\ElasticsuiteVirtualCategory\\Model\\Rule\\Condition\\Combine",
                    "conditions": [
                        {
                            "condition_type": "Smile\\ElasticsuiteVirtualCategory\\Model\\Rule\\Condition\\Product",
                            "operator": "{}",
                            "attribute_name": "sku",
                            "value": "24-MB01, 24-MB04, 24-MB03"
                        },
                        {
                            "condition_type": "Smile\\ElasticsuiteVirtualCategory\\Model\\Rule\\Condition\\Product",
                            "operator": "()",
                            "attribute_name": "color",
                            "value": [
                                "49",
                                "50",
                                "51",
                                "52"
                            ]
                        }
                    ],
                    "aggregator_type": "all",
                    "operator": null,
                    "value": "1"
                }
            }
```

**Comment by romainruaud** [comment]
Implemented in PR #1384 

format to be used when creating/updating a virtual category is the same that I showed on previous comment.

Anyway, I'm not sure it's an easy step for anyone to create such kind of complex objects through an API... But I took insipiration from the Magento SalesRule API and mimiced it.

Regards

**Comment by romainruaud** [comment]
Fixed by associated PR. Will be part of next minor 2.7 release.

Thank you for reporting it.
